src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SourcelinkTests.cs

```diff
@@ -0,0 +1,134 @@
+        Assert.True(files.Length > 0, $"Did not find sourcelink tool binary with expected filename pattern: {sourcelinkToolBinaryFilenamePattern}");
+
+        return files[0];
+    }
+
+    private IEnumerable<string> GetAllSymbolsPackages()
+    {
+        /*
+            At the moment we validate sourcelinks from runtime symbols package.
+            The plan is to make symbols, from all repos, available in source-build artifacts.
+            Once that's available, this code will be modified to validate all available symbols.
+            Tracking issue: https://github.com/dotnet/source-build/issues/3612
+        */
+
+        // Runtime symbols package lives in the same directory as PSB artifacts.
+        // i.e. <repo-root>/artifacts/x64/Release/runtime/dotnet-runtime-symbols-fedora.36-x64-8.0.0-preview.7.23355.7.tar.gz
+        string runtimeSymbolsPackageNamePattern = "dotnet-runtime-symbols-*.tar.gz";
+        string[] files = Directory.GetFiles(Path.GetDirectoryName(Config.SourceBuiltArtifactsPath), runtimeSymbolsPackageNamePattern, SearchOption.AllDirectories);
+        Assert.True(files.Length > 0, "Did not find runtime symbols archive");
+
+        yield return files[0];
+    }
+
+    /// <summary>
+    /// Extracts packages to subdirectories of the common symbols root directory.
+    /// </summary>
+    /// <returns>Path to common symbols root directory.</returns>
+    private string ExtractPackages(IEnumerable<string> packages)
+    {
+        string symbolsRoot = Directory.CreateDirectory(Path.Combine(SourcelinkRoot, "symbols")).FullName;
+
+        foreach (string package in packages)
+        {
+            Assert.True(package.EndsWith(".tar.gz"), $"Package extension is not supported: {package}");
+            DirectoryInfo targetDirInfo = Directory.CreateDirectory(Path.Combine(symbolsRoot, Path.GetFileNameWithoutExtension(package)));
+            Utilities.ExtractTarball(package, targetDirInfo.FullName, OutputHelper);
+        }
+
+        return symbolsRoot;
+    }
+
+    private void ValidateAllFiles(string path)
+    {
+        Assert.True(Directory.Exists(path), $"Path, with symbol files to validate, does not exist: {path}");
+
+        IEnumerable<string> allFiles = Directory.GetFiles(path, "*.pdb", SearchOption.AllDirectories);
+        Parallel.ForEach(allFiles, file =>
+        {
+            (Process Process, string StdOut, string StdErr) executeResult = ExecuteHelper.ExecuteProcess(
+                DotNetHelper.DotNetPath,
+                $"{sourcelinkToolPath} test --offline {file}",
+                OutputHelper,
+                logOutput: false,
+                logInfo: false, // Do not log process cmd line, as there can be 1,000+
+                millisecondTimeout: 5000,
+                configureCallback: (process) => DotNetHelper.ConfigureProcess(process, null));
+
+            if (executeResult.Process.ExitCode != 0)
+            {
+                failedFiles.Add(file);
```

Comment:
`failedFiles` should just be a local variable that gets returned by the `ValidateAllFiles` method.

Reply:
Indeed - it was like that originally. I got carried away with some refactoring.

Reply:
Fixed with https://github.com/dotnet/installer/pull/17309/commits/9ff1e023cececde9d7f33c7e5d4d9e093347c249